HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/activities/ExecuteActivity.kt

```diff
@@ -334,15 +308,99 @@ class ExecuteActivity : BaseActivity(), Entrypoint {
             Completable.error(UnsupportedFeatureException())
         }
 
-    private fun executeWithActions(): Completable =
-        Completable
-            .fromAction {
-                if (shouldFinishImmediately()) {
-                    finishWithoutAnimation()
+    private fun checkWifiNetworkSsid(): Completable =
+        if (shortcut.ssid.isEmpty()) {
+            Completable.fromAction() {
+                finishActivityIfNeeded()
+            }
+        } else {
+            showWifiDialogIfNeeded()
+        }
+
+    private fun showWifiDialogIfNeeded(): Completable = Completable.create { emitter ->
+        if (getCurrentSsid() == shortcut.ssid) {
+            finishActivityIfNeeded()
+            emitter.onComplete()
+        } else {
+            showWifiPickerConfirmation(emitter)
+        }
+    }
+
+    private fun getCurrentSsid(): String {
+        val wifiManager = getSystemService(WIFI_SERVICE) as WifiManager
```

Comment:
This line results in a warning in Android Studio, which suggests that `getSystemService` should be invoked from the `applicationContext`, not the activity context. As such, it should be changed to

```kotlin
val wifiManager = applicationContext.getSystemService(WIFI_SERVICE) as WifiManager
```